databuilder/query_language.py

```diff
@@ -316,6 +316,14 @@ def to_first_of_year(self):
     def to_first_of_month(self):
         return _apply(qm.Function.ToFirstOfMonth, self)
 
+    def is_between(self, start, end):
+        assert end > start, "end date must be greater than start date"
```

Comment:
Sorry, I wasn't particularly clear in my last comment about error checking. I meant that we _shouldn't_ add the error checking you've added because we can't do it in the general case. The `assert` here won't work if either of `start` or `end` are ehrQL series rather than a static date e.g if someone writes:
```py
events.date.is_between(covid_vaxx_date, covid_vaxx_date + days(14))
```

As an aside, `assert` wouldn't be quite the right thing to use for user-facing errors: `raise ValueError(...)` would probably be the most appropriate standard exception.

In general, assertions are there to catch mistakes made by developers when working on the code and not mistakes made by users of the code. (Python even has an "optimised" mode where it ignores assertions completely.) Where we want to report an error to the user we should raise on exception.

Reply:
Righto, understood. I'll remove that commit